test/e2e/must_gather_operator_tests.go

```diff
@@ -1,114 +1,1186 @@
+				pod := &corev1.Pod{}
+				_ = nonAdminClient.Get(testCtx, client.ObjectKey{
+					Name:      gatherPod.Name,
+					Namespace: testNamespace,
+				}, pod)
+				return pod.Spec.NodeName
+			}).WithTimeout(2*time.Minute).WithPolling(5*time.Second).ShouldNot(BeEmpty(),
+				"Pod should be scheduled to a node")
+
+			ginkgo.By("Monitoring Pod execution phase")
+			Eventually(func() corev1.PodPhase {
+				pod := &corev1.Pod{}
+				err := nonAdminClient.Get(testCtx, client.ObjectKey{
+					Name:      gatherPod.Name,
+					Namespace: testNamespace,
+				}, pod)
+				if err != nil {
+					return corev1.PodUnknown
+				}
+				return pod.Status.Phase
+			}).WithTimeout(2*time.Minute).WithPolling(5*time.Second).Should(
+				Or(Equal(corev1.PodRunning), Equal(corev1.PodSucceeded)),
+				"Pod should reach Running or Succeeded state")
+
+			ginkgo.By("Verifying gather container is collecting data")
+			Eventually(func() bool {
+				pod := &corev1.Pod{}
+				err := nonAdminClient.Get(testCtx, client.ObjectKey{
+					Name:      gatherPod.Name,
+					Namespace: testNamespace,
+				}, pod)
+				if err != nil {
+					return false
+				}
+
+				for _, cs := range pod.Status.ContainerStatuses {
+					if cs.Name == "gather" {
+						ginkgo.GinkgoWriter.Printf("Gather container - Ready: %v, RestartCount: %d\n",
+							cs.Ready, cs.RestartCount)
+
+						// Container should be running and not crash-looping
+						if cs.RestartCount > 2 {
+							return false
+						}
+						// Return true if container has started (even if not ready yet, data collection may be in progress)
+						if cs.State.Running != nil || cs.State.Terminated != nil {
+							return true
+						}
+					}
+				}
+				return false
+			}).WithTimeout(2*time.Minute).WithPolling(5*time.Second).Should(BeTrue(),
+				"Gather container should be running and collecting data without excessive restarts")
+
+			ginkgo.By("Waiting for Job to complete")
+			Eventually(func() int32 {
+				_ = nonAdminClient.Get(testCtx, client.ObjectKey{
+					Name:      mustGatherName,
+					Namespace: testNamespace,
+				}, job)
+				return job.Status.Succeeded + job.Status.Failed
+			}).WithTimeout(2*time.Minute).WithPolling(5*time.Second).Should(
+				BeNumerically(">", 0), "Job should eventually complete")
+
+			ginkgo.By("Verifying MustGather CR status is updated")
+			fetchedMG := &mustgatherv1alpha1.MustGather{}
+			Eventually(func() string {
+				_ = nonAdminClient.Get(testCtx, client.ObjectKey{
+					Name:      mustGatherName,
+					Namespace: testNamespace,
+				}, fetchedMG)
+				return fetchedMG.Status.Status
+			}).WithTimeout(2*time.Minute).WithPolling(5*time.Second).ShouldNot(BeEmpty(),
+				"MustGather status should be updated by operator")
+
+			ginkgo.GinkgoWriter.Printf("MustGather Status: %s - Completed: %v - Reason: %s\n",
+				fetchedMG.Status.Status, fetchedMG.Status.Completed, fetchedMG.Status.Reason)
+		})
+
+		ginkgo.It("non-admin user should be able to monitor MustGather progress", func() {
```

Comment:
instead we should look at improving the progress status before GA and let the non-admin user poll the conditions in the MustGather CR to get aware about the running state.